### PR TITLE
Enable sqlite/bundle for diesel cli

### PR DESCRIPTION
**Pull request by simonhdickson**
Trying to install the CLI on windows can be problematic, when creating a project you can just import the libsqlite3-sys feature but you can't do that when installing the cli tools. With this change you can install on Windows (or other platforms without an easy way of installing sqlite3).

Installation can be achieved using this: `cargo install diesel_cli --no-default-features --features sqlite-bundled`

### CODE REVIEW

File: diesel_cli/Cargo.toml [hunk 1]
```diff
@@ -21,6 +21,7 @@ diesel = { version = "0.16.0", default-features = false }
 dotenv = ">=0.8, <0.11"
 diesel_infer_schema = "0.16.0"
 clippy = { optional = true, version = "=0.0.162" }
+libsqlite3-sys = { version = ">=0.8.0, <0.9.0", default-features = false, optional = true, features = ["bundled"] }
```
Comment (sgrif) [inline diesel_cli/Cargo.toml]: This includes the `bundled` feature by default. I'm fine with giving a flag for this, but I don't think it should be the default.

**Comment by simonhdickson** [comment]
Is this better? Also should I add this to the appveyor script?

**Comment by sgrif** [comment]
Appveyor is already too slow, but I do think it's worth adding an additional matrix entry for this on Travis. I would like to hear input from @killercup and @Eijebong on that though.

**Comment by sgrif** [comment]
@simonhdickson Are you still interested in working on this? This needs to be rebased and added to CI before it can be merged.

**Comment by simonhdickson** [comment]
Yes will rebase, should just add it to Travis then?

**Comment by sgrif** [comment]
Yes, I'd go ahead and add a manual matrix entry that only runs the CLI tests, and enables this feature (same way we do clippy/rustfmt)

File: .travis.yml [hunk 2]
```diff
@@ -64,6 +64,10 @@ matrix:
         cargo install rustfmt-nightly --vers 0.3.1
       fi
     - cargo fmt --all -- --write-mode=diff
+  - rust: nightly
```
Comment (sgrif) [inline .travis.yml]: Can you add an env var so we know what this is, like the other items?

File: .travis.yml [hunk 3]
```diff
@@ -64,6 +64,10 @@ matrix:
...
+    script:
+    - (cd diesel_cli && cargo rustc --no-default-features --features "sqlite-bundled")
```
Comment (sgrif) [inline .travis.yml]: This needs to actually run the tests not just compile

**Comment by nickbabcock** [comment]
What needs to get this PR over the hump and merged? 😄

File: diesel_cli/Cargo.toml [hunk 4]
```diff
@@ -23,6 +23,7 @@ infer_schema_internals = { version = "1.1.0" }
 clippy = { optional = true, version = "=0.0.174" }
 migrations_internals = { version = "1.1.0" }
 url = { version = "1.4.0", optional = true }
+libsqlite3-sys = { version = ">=0.8.0, <0.9.0", default-features = false, optional = true, features = ["bundled"] }
```
Comment (sgrif) [inline diesel_cli/Cargo.toml]: You should not specify the `bundled` feature here.

**Comment by nickbabcock** [comment]
After applying the following patch:

```diff
--- a/diesel_cli/Cargo.toml
+++ b/diesel_cli/Cargo.toml
@@ -23,7 +23,7 @@ infer_schema_internals = { version = "1.1.0" }
 clippy = { optional = true, version = "=0.0.185" }
 migrations_internals = { version = "1.1.0" }
 url = { version = "1.4.0", optional = true }
-libsqlite3-sys = { version = ">=0.8.0, <0.9.0", default-features = false, optional = true, features = ["bundled"] }
+libsqlite3-sys = { version = ">=0.8.0, <0.10.0", optional = true, features = ["min_sqlite_version_3_7_16"] }

 [dev-dependencies]
 difference = "1.0"
```

I can confirm that I was able to install `diesel_cli` on a clean windows machine with `--no-default-features --features "sqlite-bundled"` and my projects using sqlite and diesel migrated fine. Btw, I copied the `libsqlite3-sys` from diesel's Cargo.toml for consistency, but I'm not sure if this is the desired form.

**Comment by kjeremy** [comment]
This is a killer on windows.  What needs to be done?

**Comment by sgrif** [comment]
The comments that have been left on this PR need to be addressed.

**Comment by nickbabcock** [comment]
Applying [the diff](https://github.com/diesel-rs/diesel/pull/1302#issuecomment-378414844) I posted earlier to this PR addresses the last outstanding comment (the environment var looks to be already taken care of). If desired, I can simply open a new PR with applied diff to this PR.

**Comment by sgrif** [comment]
Feel free to open a new PR, please leave the original author's commits in
tact if you do

**Comment by sgrif** [comment]
#1672 finished this

**Pull request closed without merging**